lib/__tests__/standalone-cache.test.js

```diff
@@ -0,0 +1,146 @@
+// make sure config doesn't change between runs.
+function getConfig() {
+  return {
+    files: `${fixturesPath}/cache/*.css`,
+    config: {
+      rules: { "block-no-empty": true, "color-no-invalid-hex": true },
+    },
+    cache: true,
+  }
+}
+
+describe("standalone cache", () => {
+  const expectedCacheFilePath = `${cwd}/.stylelintcache`
+
+  beforeEach(() => {
+    // Initial run to warm up the cache
+    return standalone(getConfig())
+  })
+
+  afterEach(() => {
+    // Clean up after each test case
+    fsExtra.removeSync(expectedCacheFilePath)
+    fsExtra.removeSync(changedFile)
+  })
+
+  it("cache file is created at $CWD/.stylelintcache", () => {
+    // Ensure cache file exists
+    expect(fileExists(expectedCacheFilePath)).toBe(true)
+    const cacheFile = fsExtra.readJsonSync(expectedCacheFilePath)
+    // Ensure cache file contains only linted css file
+    expect(typeof cacheFile[validFile] === "object").toBe(true)
+    expect(typeof cacheFile[changedFile] === "undefined").toBe(true)
+  })
+
+  it("only changed files are linted", () => {
+    // Add "changed" file
+    fsExtra.copySync(validFile, changedFile)
+    // Next run should lint only changed files
+    return standalone(getConfig()).then(output => {
+      // Ensure only changed files are linted
+      const isValidFileLinted = !!output.results.find(file => file.source === validFile)
+      const isChangedFileLinted = !!output.results.find(file => file.source === changedFile)
+      expect(isValidFileLinted).toBe(false)
+      expect(isChangedFileLinted).toBe(true)
+      // Ensure cache file contains linted css files
+      const cachedFiles = fsExtra.readJsonSync(expectedCacheFilePath)
+      expect(typeof cachedFiles[validFile] === "object").toBe(true)
+      expect(typeof cachedFiles[changedFile] === "object").toBe(true)
+    })
+  })
+
+  it("all files are linted on config change", () => {
+    fsExtra.copySync(validFile, changedFile)
+    const changedConfig = getConfig()
+    changedConfig.config.rules["block-no-empty"] = false
+    // All file should be re-linted as config has changed
+    return standalone(changedConfig).then(output => {
+      // Ensure all files are re-linted
+      const isValidFileLinted = !!output.results.find(file => file.source === validFile)
+      const isChangedFileLinted = !!output.results.find(file => file.source === changedFile)
+      expect(isValidFileLinted).toBe(true)
+      expect(isChangedFileLinted).toBe(true)
+    })
+  })
+
+  it("invalid files are not cached", () => {
+    fsExtra.copySync(invalidFile, changedFile)
+    // Should lint only changed files
+    return standalone(getConfig()).then((output) => {
+      expect(output.errored).toBe(true)
+      // Ensure only changed files are linted
+      const isValidFileLinted = !!output.results.find(file => file.source === validFile)
+      const isInvalidFileLinted = !!output.results.find(file => file.source === changedFile)
```

Comment:
I realized that changedFile is a confusing var name here, it should be something like `newFileDest`. The idea here is to create new file so standalone should process both cached and uncached file.